db/models.py

```diff
@@ -14,3 +14,43 @@ class Actor(models.Model):
 
     def __str__(self) -> str:
         return f"{self.first_name} {self.last_name}"
+
+
+class Movie(models.Model):
+    title = models.CharField(max_length=255)
+    description = models.TextField()
+    actors = models.ManyToManyField(Actor,
+                                    related_name="movies")
+    genres = models.ManyToManyField(Genre,
+                                    related_name="movies")
+
+    def __str__(self) -> str:
+        return self.title
+
+
+class CinemaHall(models.Model):
+    name = models.CharField(max_length=255)
+    rows = models.IntegerField()
+    seats_in_row = models.IntegerField()
+
+    @property
+    def capacity(self) -> int:
+        return self.rows * self.seats_in_row
+
+    def __str__(self) -> str:
+        return self.name
+
+
+class MovieSession(models.Model):
+    show_time = models.DateTimeField()
+    cinema_hall = models.ForeignKey(CinemaHall,
+                                    on_delete=models.SET_NULL,
+                                    null=True,
+                                    related_name="movie_session")
+    movie = models.ForeignKey(Movie,
+                              on_delete=models.SET_NULL,
+                              null=True,
+                              related_name="movie_session")
```

Comment:
do not agree, why do you use SET_NULL here? You should remove movie_sessions if there are no hall or movie anymore